client/state/jetpack-connect/reducer.js

```diff
@@ -118,7 +118,7 @@ export function jetpackConnectAuthorize( state = {}, action ) {
 			return Object.assign( {}, state, { autoAuthorize: false } );
 		case SERIALIZE:
 		case DESERIALIZE:
-			return {};
+			return state;
```

Comment:
I think we should, because if we don't, we lose the information when someone switches between wpcom accounts, or goes from logged out to logged in.  
If we do decide to persist, we should clear the state after authorization as well.